app/src/main/java/io/github/droidkaigi/confsched2017/view/activity/SplashActivity.java

```diff
@@ -35,13 +53,34 @@ protected void onCreate(Bundle savedInstanceState) {
     @Override
     protected void onStart() {
         super.onStart();
-        handler.postDelayed(moveActivityRunnable, 3000);
+        loadSessionsForCache();
     }
 
     @Override
     protected void onStop() {
         super.onStop();
-        handler.removeCallbacks(moveActivityRunnable);
+        compositeDisposable.dispose();
+    }
+
+    private void loadSessionsForCache() {
+        Disposable disposable = Single.zip(sessionsRepository.findAll(Locale.getDefault()),
+                mySessionsRepository.findAll(),
+                delaySingleSource(),
+                (sessions, mySessions, obj) -> Observable.empty())
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doFinally(() -> {
+                    if (isFinishing()) return;
+                    startActivity(MainActivity.createIntent(SplashActivity.this));
+                })
+                .subscribe();
```

Comment:
Actually, `SessionsFragment` has not handled error yet 🙇 
I think it's better to show "Reload" button or something on `fragment_sessions.xml`.